### PR TITLE
[Core] Extending inline docs for `openai_utils.py`, boosting coverage in `test_utils.py`, fixing #762

### PR DESCRIPTION
**Pull request by maxim-saplin**
## Why are these changes needed?

- Improved docstrings for all config_list creation functions declared in `openai_utils.py`
- `test_utils.py` has more cases tested ramping up coverage
- `test_utils.py` no longer depends on files that are part of .gitignore (and hence not downloaded when cloning), utilises temp files and fake data
- `openai_utils.get_config_list()` has an assertion verifying that number of items in files is equal (replacing the current behaviour with out of index being thrown)

## Related issue number

Closes https://github.com/microsoft/autogen/issues/762

## Checks

- [x] (NOT RELEVANT) I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.
         _P.S. settings up the full suite with all required open configs is a challenge, though I made sure the parts and tests impacted by the changes ran fine_

### CODE REVIEW

File: autogen/oai/openai_utils.py [hunk 1]
```diff
@@ -65,14 +65,36 @@ def get_key(config):
...
 
     Args:
         api_keys (list): The api keys for openai api calls.
-        base_urls (list, optional): The api bases for openai api calls.
+        base_urls (list, optional): The api bases for openai api calls. If provided, should match the length of api_keys.
         api_type (str, optional): The api type for openai api calls.
         api_version (str, optional): The api version for openai api calls.
+
+    Returns:
+        list: A list of configs for OepnAI API calls.
```
Comment (ekzhu) [inline autogen/oai/openai_utils.py:77]: list: A list of configs for OpenAI API calls.

**Review by ekzhu — APPROVED** [review]

**Comment by ekzhu** [comment]
Thanks for the PR! Could you take a look at the missing code coverage for the changes made and also a small typo?

**Review by ekzhu — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you take a look at the missing code coverage for the changes made and also a small typo?

File: autogen/oai/openai_utils.py (hunk 1 above)
Comment (maxim-saplin) [reply autogen/oai/openai_utils.py:77]: Looks good!

**Comment by maxim-saplin** [comment]
Away for keyboard will check the coverage tomorrow. Thanks!

**Comment by maxim-saplin** [comment]
@ekzhu - I've fixed Codecov warnings, should be good

File: autogen/oai/openai_utils.py [hunk 2]
```diff
@@ -101,20 +123,56 @@ def config_list_openai_aoai(
     key_file_path: Optional[str] = ".",
     openai_api_key_file: Optional[str] = "key_openai.txt",
     aoai_api_key_file: Optional[str] = "key_aoai.txt",
+    openai_api_base_file: Optional[str] = "base_openai.txt",
```
Comment (sonichi) [inline autogen/oai/openai_utils.py:126]: The docstr for this var is missing.

Comment (ekzhu) [reply autogen/oai/openai_utils.py:126]: I added it.

File: test/oai/test_utils.py [hunk 3]
```diff
@@ -5,11 +5,9 @@
 import logging
 import tempfile
 from unittest import mock
+from unittest.mock import patch
 import autogen  # noqa: E402
 
-KEY_LOC = "notebook"
-OAI_CONFIG_LIST = "OAI_CONFIG_LIST"
```
Comment (ekzhu) [inline test/oai/test_utils.py:11]: We are currently trying to improve the test utils by having a centralized way to obtain the configs. @afourney do you think we should keep these variables in test_utils.py for now?

**Review by ekzhu — COMMENTED** [review]
Ok

File: test/oai/test_utils.py (hunk 3 above)
Comment (maxim-saplin) [reply test/oai/test_utils.py:11]: Specifically in `test_utils.py` those 2 file names are no longer needed, all is covered by mocks/fakes, the tests do not depend on external model configs.

Speaking of consolidating and tiding up models used by tests, I have already added `--skip-openai` flag to `pytest`: https://github.com/microsoft/autogen/pull/1097

The next step could be smth like this: https://github.com/microsoft/autogen/pull/1052#discussion_r1436008853

If there're any other ideas/issues in regards to this (besides #1089) - please share

Comment (ekzhu) [reply test/oai/test_utils.py:11]: Got it. Thanks!

**Review by ekzhu — APPROVED** [review]

**Pull request merged**